scripts/generate_spherical_expansion_ref_data.py

```diff
@@ -64,6 +64,7 @@ def dump_reference_json():
                 max_radials=max_radials,
                 rep_info=[])
 
+    # An example of the gruesomeness of using 4 spaces for one tab
```

Comment:
More like, the gruesomeness of using so many levels of nested indentation. To quote (only half-jokingly) the Linux kernel coding conventions:
> Now, some people will claim that having 8-character indentations makes the code move too far to the right, and makes it hard to read on a 80-character terminal screen. The answer to that is that if you need more than 3 levels of indentation, you’re screwed anyway, and should fix your program.

(source: https://www.kernel.org/doc/html/v4.10/process/coding-style.html)

Reply:
https://docs.python.org/3/library/itertools.html#itertools.product (for the first comment)

Reply:
the point of these reference data is to fix the output of the code. At some point someone checked the correctness of the output and then fixed it so that a future modification that break the feature will be detected right away.

Reply:
I'm still not sure I understand... eventually we want to compare with a tested implementation like the python version of SOAP. And another thing, what's the point of `scripts/soap_example.py` and `scripts/spherical_expansion_example.py`? They look identical to their `generate_<rep>_ref_data.py` counterparts... so can they be deleted?